modules/openapi-generator/src/main/resources/ruby-client/api.mustache

```diff
@@ -175,20 +175,38 @@ module {{moduleName}}
       # http body (model)
       post_body = opts[:debug_body]{{#bodyParam}} || @api_client.object_to_http_body({{#required}}{{{paramName}}}{{/required}}{{^required}}opts[:'{{{paramName}}}']{{/required}}){{/bodyParam}}
 
-      # return_type
-      return_type = opts[:debug_return_type]{{#returnType}} || '{{{returnType}}}'{{/returnType}}
-
       # auth_names
       auth_names = opts[:debug_auth_names] || [{{#authMethods}}'{{name}}'{{#hasMore}}, {{/hasMore}}{{/authMethods}}]
 
+      # return_type
+      return_type = opts[:debug_return_type]
+
+      {{#returnType}}
+      {{#responses}}
+      {{#-first}}
+      return_types_map = {
+      {{/-first}}
+        {{^isWildcard}}
+        {{code}} => '{{dataType}}',
+        {{/isWildcard}}
+      {{#-last}}
+      }
+      {{/-last}}
+      {{/responses}}
+      {{/returnType}}
+      {{^returnType}}
+      return_types_map = {}
+      {{/returnType}}
+
       new_options = opts.merge(
         :operation => :"{{classname}}.{{operationId}}",
         :header_params => header_params,
         :query_params => query_params,
         :form_params => form_params,
         :body => post_body,
         :auth_names => auth_names,
-        :return_type => return_type
+        :return_type => return_type,
+        :return_types_map => return_types_map
```

Comment:
You are using two parameters to pass in the same data here:
- return_type
- return_types_map
How about only passing in return_types_map?
If the user sets a :debug_return_type should we set all values of the map to  :debug_return_type as it looks the old code replace the return type with the debug_return_type if it is passed in.